### PR TITLE
52017 - Change query to retrieve Taken with identificatie

### PR DESCRIPTION
**Pull request by theo-ritense**

### CODE REVIEW

File: gzac/objects-api/src/test/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/graphql/TaskMutationIT.kt [hunk 1]
```diff
@@ -99,11 +100,11 @@ internal class TaskMutationIT(
...
             override fun dispatch(request: RecordedRequest): MockResponse {
-                val path = request.path?.substringBefore('?')
-                val queryParams = request.path?.substringAfter('?')?.split('&') ?: emptyList()
-                val response = when (request.method + " " + path) {
+                val uri = UriComponentsBuilder.fromUri(request.requestUrl!!.toUri()).build()
+                val response = when (request.method + " " + uri.path) {
                     "GET /api/v2/objects" -> {
-                        if (queryParams.any { it.contains("bsn__exact__569312863") }) {
+                        val dataAttrs = uri.queryParams["data_attrs"]!!.first().split(",")
+                        if (dataAttrs.containsAll(listOf("identificatie.type__exact__bsn", "identificatie.value__exact__569312863"))) {
```
Comment (Klaas-Ritense) [inline gzac/objects-api/src/test/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/graphql/TaskMutationIT.kt:107]: According to the [API specs](https://objecten-zgw.test.denhaag.nl/api/v2/schema/#tag/objects/operation/object_list), a nested object can be queried like this:
`identificatie__type__exact__bsn`

Comment (theo-ritense) [reply gzac/objects-api/src/test/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/graphql/TaskMutationIT.kt]: I missed that part. Fixed!

File: gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/service/TaskService.kt [hunk 2]
```diff
@@ -79,22 +79,33 @@ open class TaskService(
...
+    private fun getUserSearchParameters(authentication: CommonGroundAuthentication): List<ObjectSearchParameter> {
         return when (authentication) {
-            is BurgerAuthentication -> ObjectSearchParameter("bsn", Comparator.EQUAL_TO, authentication.getBsn())
-            is BedrijfAuthentication -> ObjectSearchParameter("kvk", Comparator.EQUAL_TO, authentication.getKvkNummer())
+            is BurgerAuthentication -> {
+                createIdentificatieSearchParameters("bsn", authentication.getBsn())
+            }
+            is BedrijfAuthentication -> {
+                createIdentificatieSearchParameters("kvk", authentication.getKvkNummer())
+            }
```
Comment (Klaas-Ritense) [inline gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/service/TaskService.kt:100]: This is a massive breaking change. The existing TaakConnector, will stop working. And there are several projects that use the TaakConnector in production.
Have we considered what we want to do with this

Comment (theo-ritense) [reply gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/service/TaskService.kt:100]: @Klaas-Ritense Checked with Thomas, and it seems there 'is just 1 user of which none in production'.

**Review by Klaas-Ritense — CHANGES_REQUESTED** [review]
The 'identificatie__type' is also inside an array. Not sure if that works

**Comment by theo-ritense** [comment]
That's an issue. It not in the spec, so we should expect it not to work.
@ThomasRitense can we change the TaakObject.identification to an object instead of a list/array?

**Comment by ThomasRitense** [comment]
@theo-ritense Yes, please change from array to object.

**Review by Klaas-Ritense — APPROVED** [review]

**Pull request merged**